subsys/bluetooth/controller/Kconfig

```diff
@@ -470,20 +470,17 @@ config BT_CTLR_CHAN_SEL_2
 	  the Controller.
 
 config BT_CTLR_ADV_EXT
-	bool "LE Advertising Extensions" if !BT_LL_SW_SPLIT
+	bool "LE Advertising Extensions"
```

Comment:
Wouldn't this still result in multiple entries of `BT_CTLR_ADV_EXT` when you search for it?

Reply:
Not sure about the search, but it will definitely break menuconfig. The rendering logic that produces the current duplicate entry thing in menuconfig is caused by menuconfig forcing menus with visible children into existence, even if they should be invisible. As the menu parent is always the last node before the `if`, the `[EXPERIMENTAL]` node is forced into existence, as it holds all the child nodes.

IMO, the right variant of this suggestion would be to keep the two entries, but remove the prompts for both, then create a single prompt entry below them that conditionally adds both prompts:

```
config BT_CTLR_ADV_EXT
    prompt "LE Advertising Extensions [EXPERIMENTAL]" if BT_LL_SW_SPLIT
    prompt "LE Advertising Extensions" 
```

This entry will only serve as a toggleable menu heading, really, but it should keep the logic of the other properties as-is.

Reply:
> Wouldn't this still result in multiple entries of `BT_CTLR_ADV_EXT` when you search for it?

There will be multiple entries when searched, but only one is enabled based on the `if` condition that is satisfied. This is similar to how all hidden Kconfig get listed when in search.

My suggestions do not have the original problem of duplicate entries when traversing using menuconfig/guiconfig.

> the right variant of this suggestion would be to keep the two entries, but remove the prompts for both, then create a single prompt entry below them that conditionally adds both prompts

This (multiple conditional prompt under an entry) is not permitted by Kconfig and fail to build with a warning.

Reply:
So @cvinayak , @alwa-nordic, and @trond-snekvik . The remaining question is: Are we ok with having two entries? It looks a bit messy IMO. But the proposed solution is better than the existing one

Reply:
It is ok to have see two prompt messages, and is consistent with the Link Layer selection wherein only one is selectable based on the Link Layer enabled.